### ISSUE TITLE
Add the ability to disable a mod from updating when choosing "update all"

### ISSUE THREAD

**Issue by Corvus20** [gorilla-devs/GDLauncher#1020]
**Is your feature request related to a problem? Please describe.**
While it is a godsend to update all of my mods at the click of a button, this can occasionally break them. Some mods are incompatible with another mod of a certain version, requiring one to stay on a previous version, or a new version simply has some game breaking issue that you don't want to play with. or in my case a mod has a split release with different versions that are built differently to be compatible with a certain mod if it is present. This causes a problem, either update all the mods and manually re-install the older versions of the mods you didn't want to update, or manually update each mod that you want updated and skip over the ones you don't.

**Describe the solution you'd like**
A simple toggleable "update lock" switch that will enable/disable the Update All button from updating that specific mod.

**Describe alternatives you've considered**
I would either update all the mods and manually re-install the older versions of the mods you didn't want to update, or manually update each mod that you want updated and skip over the ones you don't. Neither are very efficient or practical.

**Additional context**
This is the only gripe I have with GDLauncher right now, aside from that I love it! Keep up the good work!

**Comment by Eskaan** [comment]
Do we really need that? I mean when you don’t want an Mod to update just do the others manually

**Comment by Corvus20** [comment]
Well at least from how I am looking at it, it would be easier to just lock the mods you dont want updated and then press update all to update all the unlocked ones, as opposed to having to scroll through and update every single one manually.

**Comment by Eskaan** [comment]
Your Project when you accept every feature request
![credit to someone on r/programmerhumor](https://preview.redd.it/v6g43ll55ih71.jpg?width=640&crop=smart&auto=webp&s=f5b574bba64206a4e2f458170887a4f4a42a13f3)

**Comment by Corvus20** [comment]
Its not like I am making any extremely outlandish requests though, and isn't the point of having a GitHub page like this getting feedback from users in regards to problems or potential improvements? I know you can't accept every feature request, and I don't expect you to do so, I am just providing a suggestion for a possible improvement, and whether or not you want to implement it is ultimately up to you.

I'm just suggesting this because it would be a small, convenient quality of life feature that would save a ton clicking and scrolling, and I think would go a long way in terms of managing mods, and I don't see what problems would arise from implementing something like it.

**Comment by blarfoon** [comment]
I understand the need behind this request, it's actually very similar to how I update nodejs dependencies.
I don't like the idea of just adding a lock update for a mod, it needs to be a more extensive update on how mods update works.
I am currently on vacation but when I come back I will try to create a few designs on how this could work.

**Comment by kreezxil** [comment]
I agree with @Corvus20 on this one. The **update-all** feature is nice and would be even nicer if we had this **update-lock** option. 

Case in point, my latest 1.16.5 packs in development will typically have about 40 or more mods that need updating each week. But some mods such as Citadel for Ice and Fire need to be prevented from updating to keep Ice and Fire from breaking. From time to time, even Minecolonies is like this, requiring an older version of Structurize. I've encountered this need with the FTB suite of mods too. Occasionally, you don't want their latest and greatest because of breakage. And yet, to be able to easily hit **update-all**, well, that's my crack, and I'm loving it. :)

**Comment by blarfoon** [comment]
I agree this is something nice to have. I a not sure whether or not it's doable in the current design but we will surely keep this in mind when working on the redesign/rewrite

**Comment by Eskaan** [comment]
Think its possible. Probably needs a .json file to store the data or a [...]-noUpdate[.jar] extension on the mod file

**Comment by Velken** [comment]
a recent case use for this is minecolonies/structurize for 1.18.2
minecolonies requires a version that is not the lastest of structurize
dev of minecolonies said they wont update because there were big changes on structurize
